PR: https://github.com/odoo/odoo/pull/208005

From: 2fca445d07a803953481e8b4967806e3453a00da
From: Maxime de Neuville
Date: 2025-09-09 12:52:48

Breaking data model changes scores: del:5 + add:0, change matches:
-    disabled_rating_warning = fields.Text(compute='_compute_disabled_rating_warning', export_string_translation=False)
-    group_project_rating = fields.Boolean("Customer Ratings", implied_group='project.group_project_rating')
-    group_project_recurring_tasks = fields.Boolean("Recurring Tasks", implied_group="project.group_project_recurring_tasks", group='base.group_portal,base.group_user')
-    group_project_task_dependencies = fields.Boolean("Task Dependencies", implied_group="project.group_project_task_dependencies")
-    group_project_milestone = fields.Boolean('Milestones', implied_group='project.group_project_milestone', group='base.group_portal,base.group_user')

Total Changes: 265

[IMP] project,sale_project,sale_timesheet,web: clean settings

This commit aims to remove some global settings to only keep them in the project form view:

1. More specifically, it removes the global settings "Recurring Tasks", "Task Dependencies" and "Milestones" from the global settings.
And moves "Recurring Tasks" and "Task Dependencies" in the project form view directly. It keeps the "Milestones" settings in the project form view as it was.

* The way groups "project.group_project_milestone", "project.group_project_recurring_tasks" and "project.group_project_recurring_tasks" are given to the user is now different, as the global settings are removed.
It now works like this:
=> When the user enables one of the settings we mentioned (e.g. "Milestones") in any project and the related group (e.g. "group_project_milestone") is not already added to the base user group, it is added.
=> If the user disables the setting from a project/remove a project which had the setting enabled, if after that disabling there is no project (even archived ones) having the setting enabled, then the related group is removed from the base user group.

* The default value given to the setting of a new project is always False.

* We remove the behaviour introduced by this commit: https://github.com/odoo/odoo/commit/50ffc339991b136b48525b2d60aa34a3039bb8da

2. The "Customer Ratings" setting was removed from the global settings and the project form view. It is moved to the task stage form view.

* If the rating status of a stage is set to "stage": when a task reaches that stage, an email will be automatically sent to get the customer rating.
  If the rating status of a stage is set to "periodic": an email will be sent periodically at the given interval as long as the task remains in the stage.
* If sale_project is installed, we only show the setting if at least one of the project of the stage is billable
* The "Rating Email Template" is now required if the setting is enabled.
* Some renaming was made in the setting description to make it clearer.
* The group "project.group_project_rating" is removed so we always make visible the customer rating in all views and in the reporting.

3. When the user enables the top bar a project which only has one visible action (e.g. "Tasks"), it now directly opens the dropdown so that the user will gain time to select the actions to show.

task-4700865

closes odoo/odoo#208005

Related: odoo/enterprise#84657
Related: odoo/upgrade#7640
Signed-off-by: Vincent Larcin (vila) <vila@odoo.com>

================================= pseudo patch: =================================

--- a/addons/project/models/project_milestone.py
+++ b/addons/project/models/project_milestone.py
@@ -25,6 +25,7 @@ class ProjectMilestone(models.Model):
     is_reached = fields.Boolean(string="Reached", default=False, copy=False)
     reached_date = fields.Date(compute='_compute_reached_date', store=True, export_string_translation=False)
     task_ids = fields.One2many('project.task', 'milestone_id', 'Tasks', export_string_translation=False)
+    project_allow_milestones = fields.Boolean(compute='_compute_project_allow_milestones', search='_search_project_allow_milestones', compute_sudo=True, export_string_translation=False)
 
     # computed non-stored fields
     is_deadline_exceeded = fields.Boolean(compute="_compute_is_deadline_exceeded", export_string_translation=False)
@@ -86,6 +87,17 @@ class ProjectMilestone(models.Model):
             opened_task_count, closed_task_count = task_count_per_milestones[milestone.id]
             milestone.can_be_marked_as_done = closed_task_count > 0 and not opened_task_count
 
+    @api.depends('project_id.allow_milestones')
+    def _compute_project_allow_milestones(self):
+        for milestone in self:
+            milestone.project_allow_milestones = milestone.project_id.allow_milestones
+
+    def _search_project_allow_milestones(self, operator, value):
+        query = self.env['project.project'].sudo()._search([
+            ('allow_milestones', operator, value),
+        ])
+        return [('project_id', 'in', query)]
+
     def toggle_is_reached(self, is_reached):
         self.ensure_one()
         self.update({'is_reached': is_reached})

--- a/addons/project/models/project_project.py
+++ b/addons/project/models/project_project.py
@@ -144,8 +144,9 @@ class ProjectProject(models.Model):
     date_start = fields.Date(string='Start Date', copy=False)
     date = fields.Date(string='Expiration Date', copy=False, index=True, tracking=True,
         help="Date on which this project ends. The timeframe defined on the project is taken into account when viewing its planning.")
-    allow_task_dependencies = fields.Boolean('Task Dependencies', default=lambda self: self.env.user.has_group('project.group_project_task_dependencies'), inverse='_inverse_allow_task_dependencies')
-    allow_milestones = fields.Boolean('Milestones', default=lambda self: self.env.user.has_group('project.group_project_milestone'))
+    allow_task_dependencies = fields.Boolean('Task Dependencies', inverse='_inverse_allow_task_dependencies')
+    allow_milestones = fields.Boolean('Milestones', inverse='_inverse_allow_milestones')
+    allow_recurring_tasks = fields.Boolean('Recurring Tasks', inverse='_inverse_allow_recurring_tasks')
     tag_ids = fields.Many2many('project.tags', relation='project_project_project_tags_rel', string='Tags')
     task_properties_definition = fields.PropertiesDefinition('Task Properties')
     closed_task_count = fields.Integer(compute="_compute_closed_task_count", export_string_translation=False)
@@ -155,24 +156,6 @@ class ProjectProject(models.Model):
     collaborator_ids = fields.One2many('project.collaborator', 'project_id', string='Collaborators', copy=False, export_string_translation=False)
     collaborator_count = fields.Integer('# Collaborators', compute='_compute_collaborator_count', compute_sudo=True, export_string_translation=False)
 
-    # rating fields
-    rating_request_deadline = fields.Datetime(compute='_compute_rating_request_deadline', store=True, export_string_translation=False)
-    rating_active = fields.Boolean('Customer Ratings', default=lambda self: self.env.user.has_group('project.group_project_rating'))
-    rating_status = fields.Selection(
-        [('stage', 'when reaching a given stage'),
-         ('periodic', 'on a periodic basis')
-        ], 'Customer Ratings Status', default="stage", required=True,
-        help="Collect feedback from your customers by sending them a rating request when a task enters a certain stage. To do so, define a rating email template on the corresponding stages.\n"
-             "Rating when changing stage: an email will be automatically sent when the task reaches the stage on which the rating email template is set.\n"
-             "Periodic rating: an email will be automatically sent at regular intervals as long as the task remains in the stage in which the rating email template is set.")
-    rating_status_period = fields.Selection([
-        ('daily', 'Daily'),
-        ('weekly', 'Weekly'),
-        ('bimonthly', 'Twice a Month'),
-        ('monthly', 'Once a Month'),
-        ('quarterly', 'Quarterly'),
-        ('yearly', 'Yearly')], 'Rating Frequency', required=True, default='monthly')
-
     # Not `required` since this is an option to enable in project settings.
     stage_id = fields.Many2one('project.project.stage', string='Stage', ondelete='restrict', groups="project.group_project_stages",
         tracking=True, index=True, copy=False, default=_default_stage_id, group_expand='_read_group_expand_full')
@@ -301,12 +284,6 @@ class ProjectProject(models.Model):
                     _("The project's company cannot be changed if its analytic account has analytic lines or if more than one project is linked to it."))
             account.company_id = project.company_id or project.partner_id.company_id
 
-    @api.depends('rating_status', 'rating_status_period')
-    def _compute_rating_request_deadline(self):
-        periods = {'daily': 1, 'weekly': 7, 'bimonthly': 15, 'monthly': 30, 'quarterly': 90, 'yearly': 365}
-        for project in self:
-            project.rating_request_deadline = fields.Datetime.now() + timedelta(days=periods.get(project.rating_status_period, 0))
-
     @api.depends('last_update_id.status')
     def _compute_last_update_status(self):
         for project in self:
@@ -445,6 +422,17 @@ class ProjectProject(models.Model):
             )
         ):
             waiting_tasks.state = '01_in_progress'
+        res = self._check_project_group_with_field('allow_task_dependencies', 'project.group_project_task_dependencies')
+        # Hide/Show task waiting subtype when task dependencies feature is disabled/enabled
+        if res or res is False:
+            self.env.ref('project.mt_task_waiting').hidden = not res
+            self.env.ref('project.mt_project_task_waiting').hidden = not res
+
+    def _inverse_allow_milestones(self):
+        self._check_project_group_with_field('allow_milestones', 'project.group_project_milestone')
+
+    def _inverse_allow_recurring_tasks(self):
+        self._check_project_group_with_field('allow_recurring_tasks', 'project.group_project_recurring_tasks')
 
     @api.model
     def _map_tasks_default_values(self, project):
@@ -679,6 +667,9 @@ class ProjectProject(models.Model):
         if 'allow_task_dependencies' in vals and not vals.get('allow_task_dependencies'):
             self.env['project.task'].search([('project_id', 'in', self.ids), ('state', '=', '04_waiting_normal')]).write({'state': '01_in_progress'})
 
+        if 'allow_recurring_tasks' in vals and not vals['allow_recurring_tasks']:
+            self.env['project.task'].search([('project_id', 'in', self.ids), ('recurring_task', '=', True)]).write({'recurring_task': False})
+
         if 'active' in vals:
             # archiving/unarchiving a project does it on its tasks, too
             self.with_context(active_test=False).mapped('tasks').write({'active': vals['active']})
@@ -709,6 +700,12 @@ class ProjectProject(models.Model):
         analytic_accounts_to_delete.unlink()
         return result
 
+    @api.ondelete(at_uninstall=False)
+    def _check_project_group_at_removal(self):
+        self._check_project_group_with_field('allow_task_dependencies', 'project.group_project_task_dependencies')
+        self._check_project_group_with_field('allow_milestones', 'project.group_project_milestone')
+        self._check_project_group_with_field('allow_recurring_tasks', 'project.group_project_recurring_tasks')
+
     def _order_field_to_sql(self, alias, field_name, direction, nulls, query):
         if field_name == 'is_favorite':
             sql_field = SQL(
@@ -773,6 +770,56 @@ class ProjectProject(models.Model):
             ['id', 'name'],
         )
 
+    @api.model
+    def _check_project_group_with_field(self, field_name, group_name):
+        """ Check if the user has the group 'group_name' and if there is a project with the field 'field_name' set to True.
+        If not, remove the group 'group_name' from the user base group.
+        Otherwise, add the group 'group_name' to the user base group.
+        Returns True if the group was added, False if it was removed, None if no change was made.
+        """
+        has_user_group = bool(self.env.user.has_group(group_name))
+        group = self.env.ref(group_name)
+        base_group_user = self.env.ref('base.group_user')
+        has_project_field_set = bool(self.env['project.project'].search_count([(field_name, '=', True)], limit=1))
+        res = None
+
+        if not has_user_group and has_project_field_set:
+            # add the group to the base user group if there is at least one project with field_name=True
+            base_group_user.sudo().write({
+                'implied_ids': [Command.link(group.id)]
+            })
+            res = True
+        elif has_user_group and not has_project_field_set:
+            # remove the group from the base user group if there is no project with field_name=True
+            base_group_user.sudo().write({
+                'implied_ids': [Command.unlink(group.id)]
+            })
+            group.sudo().write({'user_ids': [Command.clear()]})
+            res = False
+        return res
+
+    def _get_project_features_mapping(self):
+        return {
+            'allow_task_dependencies': 'project.group_project_task_dependencies',
+            'allow_milestones': 'project.group_project_milestone',
+            'allow_recurring_tasks': 'project.group_project_recurring_tasks',
+        }
+
+    @api.model
+    def check_features_enabled(self, updated_features=None):
+        if not self.env.user.has_group('project.group_project_user'):
+            return {}
+        if updated_features:
+            return {
+                field_name: self.env.user.has_group(group)
+                for field_name, group in self._get_project_features_mapping().items()
+                if field_name in updated_features
+            }
+        return {
+            field_name: self.env.user.has_group(group)
+            for field_name, group in self._get_project_features_mapping().items()
+        }
+
     # ---------------------------------------------------
     # Mail gateway
     # ---------------------------------------------------
@@ -796,8 +843,6 @@ class ProjectProject(models.Model):
 
     def _mail_get_message_subtypes(self):
         res = super()._mail_get_message_subtypes()
-        if not self.rating_active:
-            res -= self.env.ref('project.mt_project_task_rating')
         if len(self) == 1:
             waiting_subtype = self.env.ref('project.mt_project_task_waiting')
             if not self.allow_task_dependencies and waiting_subtype in res:
@@ -878,6 +923,8 @@ class ProjectProject(models.Model):
             'create': self.active,
             'active_test': self.active,
             'active_id': self.id,
+            'allow_milestones': self.allow_milestones,
+            'allow_task_dependencies': self.allow_task_dependencies,
             })
         action['context'] = context
         if self.is_template:
@@ -1033,7 +1080,7 @@ class ProjectProject(models.Model):
             'show': True,
             'sequence': 1,
         }]
-        if self.rating_count != 0 and self.env.user.has_group('project.group_project_rating'):
+        if self.rating_count != 0:
             if self.rating_avg >= rating_data.RATING_AVG_TOP:
                 icon = 'smile-o text-success'
             elif self.rating_avg >= rating_data.RATING_AVG_OK:
@@ -1046,7 +1093,6 @@ class ProjectProject(models.Model):
                 'number': f'{int(self.rating_avg) if self.rating_avg.is_integer() else round(self.rating_avg, 1)} / 5',
                 'action_type': 'object',
                 'action': 'action_view_all_rating',
-                'show': self.rating_active,
                 'sequence': 15,
             })
         if self.env.user.has_group('project.group_project_user'):
@@ -1150,23 +1196,6 @@ class ProjectProject(models.Model):
             'default_company_id': unquote('company_id'),
         }
 
-    # ---------------------------------------------------
-    # Rating business
-    # ---------------------------------------------------
-
-    # This method should be called once a day by the scheduler
-    @api.model
-    def _send_rating_all(self):
-        projects = self.search([
-            ('rating_active', '=', True),
-            ('rating_status', '=', 'periodic'),
-            ('rating_request_deadline', '<=', fields.Datetime.now())
-        ])
-        for project in projects:
-            project.task_ids._send_task_rating_mail()
-            project._compute_rating_request_deadline()
-            self.env.cr.commit()
-
     # ---------------------------------------------------
     # Privacy
     # ---------------------------------------------------

--- a/addons/project/models/project_task.py
+++ b/addons/project/models/project_task.py
@@ -35,6 +35,7 @@ PROJECT_TASK_READABLE_FIELDS = {
     'user_ids',
     'display_parent_task_button',
     'current_user_same_company_partner',
+    'allow_recurring_tasks',
     'allow_milestones',
     'milestone_id',
     'has_late_and_unreached_milestone',
@@ -231,7 +232,7 @@ class ProjectTask(models.Model):
     email_cc = fields.Char(help='Email addresses that were in the CC of the incoming emails from this task and that are not currently linked to an existing customer.')
     company_id = fields.Many2one('res.company', string='Company', compute='_compute_company_id', store=True, readonly=False, recursive=True, copy=True, default=_default_company_id)
     color = fields.Integer(string='Color Index', export_string_translation=False)
-    rating_active = fields.Boolean(string='Project Rating Status', related="project_id.rating_active")
+    rating_active = fields.Boolean(string='Stage Rating Status', related="stage_id.rating_active")
     attachment_ids = fields.One2many(
         'ir.attachment',
         compute='_compute_attachment_ids',
@@ -291,6 +292,7 @@ class ProjectTask(models.Model):
     display_follow_button = fields.Boolean(compute='_compute_display_follow_button', compute_sudo=True, export_string_translation=False)
 
     # recurrence fields
+    allow_recurring_tasks = fields.Boolean(related='project_id.allow_recurring_tasks', export_string_translation=False)
     recurring_task = fields.Boolean(string="Recurrent")
     recurring_count = fields.Integer(string="Tasks in Recurrence", compute='_compute_recurring_count')
     recurrence_id = fields.Many2one('project.task.recurrence', copy=False, index='btree_not_null')
@@ -1323,7 +1325,7 @@ class ProjectTask(models.Model):
 
         # rating on stage
         if 'stage_id' in vals and vals.get('stage_id'):
-            self.sudo().filtered(lambda x: x.project_id.rating_active and x.project_id.rating_status == 'stage')._send_task_rating_mail(force_send=True)
+            self.sudo().filtered(lambda x: x.stage_id.rating_active and x.stage_id.rating_status == 'stage')._send_task_rating_mail(force_send=True)
 
         if 'state' in vals:
             # specific use case: when the blocked task goes from 'forced' done state to a not closed state, we fix the state back to waiting
@@ -1604,7 +1606,7 @@ class ProjectTask(models.Model):
 
     def _mail_get_message_subtypes(self):
         res = super()._mail_get_message_subtypes()
-        if not self.project_id.rating_active:
+        if not self.stage_id.rating_active:
             res -= self.env.ref('project.mt_task_rating')
         if len(self) == 1:
             waiting_subtype = self.env.ref('project.mt_task_waiting')

--- a/addons/project/models/project_task_type.py
+++ b/addons/project/models/project_task_type.py
@@ -2,6 +2,7 @@
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
 from collections import defaultdict
+from datetime import timedelta
 
 from odoo import api, fields, models, _
 from odoo.exceptions import UserError
@@ -38,18 +39,51 @@ class ProjectTaskType(models.Model):
         string='Rating Email Template',
         domain=[('model', '=', 'project.task')],
         help="If set, a rating request will automatically be sent by email to the customer when the task reaches this stage. \n"
-             "Alternatively, it will be sent at a regular interval as long as the task remains in this stage, depending on the configuration of your project. \n"
-             "To use this feature make sure that the 'Customer Ratings' option is enabled on your project.")
+             "Alternatively, it will be sent at a regular interval as long as the task remains in this stage.")
     auto_validation_state = fields.Boolean('Automatic Kanban Status', default=False,
         help="Automatically modify the state when the customer replies to the feedback for this stage.\n"
             " * Good feedback from the customer will update the state to 'Approved' (green bullet).\n"
             " * Neutral or bad feedback will set the kanban state to 'Changes Requested' (orange bullet).\n")
-    disabled_rating_warning = fields.Text(compute='_compute_disabled_rating_warning', export_string_translation=False)
     rotting_threshold_days = fields.Integer('Days to rot', default=0, help='Day count before tasks in this stage become stale. Set to 0 to disable \
         Changing this parameter will not affect the rotting status/date of resources last updated before this change.')
 
     user_id = fields.Many2one('res.users', 'Stage Owner', default=_default_user_id, compute='_compute_user_id', store=True, index=True)
 
+    # rating fields
+    rating_request_deadline = fields.Datetime(compute='_compute_rating_request_deadline', store=True, export_string_translation=False)
+    rating_active = fields.Boolean('Send a customer rating request')
+    rating_status = fields.Selection(
+        string='Customer Ratings Status',
+        selection=[
+            ('stage', 'when reaching this stage'),
+            ('periodic', 'on a periodic basis'),
+        ],
+        default='stage',
+        required=True,
+        help="Collect feedback from your customers by sending them a rating request when a task enters a certain stage. To do so, define a rating email template on the stage.\n"
+             "Rating when changing stage: an email will be automatically sent when a task reaches the stage.\n"
+             "Periodic rating: an email will be automatically sent at regular intervals as long as the task remains in the stage.",
+    )
+    rating_status_period = fields.Selection(
+        string='Rating Frequency',
+        selection=[
+            ('daily', 'Daily'),
+            ('weekly', 'Weekly'),
+            ('bimonthly', 'Twice a Month'),
+            ('monthly', 'Once a Month'),
+            ('quarterly', 'Quarterly'),
+            ('yearly', 'Yearly'),
+        ],
+        default='monthly',
+        required=True,
+    )
+
+    @api.depends('rating_status', 'rating_status_period')
+    def _compute_rating_request_deadline(self):
+        periods = {'daily': 1, 'weekly': 7, 'bimonthly': 15, 'monthly': 30, 'quarterly': 90, 'yearly': 365}
+        for stage in self:
+            stage.rating_request_deadline = fields.Datetime.now() + timedelta(days=periods.get(stage.rating_status_period, 0))
+
     def unlink_wizard(self, stage_view=False):
         self = self.with_context(active_test=False)
         # retrieves all the projects with a least 1 task in that stage
@@ -78,6 +112,16 @@ class ProjectTaskType(models.Model):
     def write(self, vals):
         if 'active' in vals and not vals['active']:
             self.env['project.task'].search([('stage_id', 'in', self.ids)]).write({'active': False})
+        # Hide/Show task rating template when customer rating feature is disabled/enabled
+        if 'rating_active' in vals:
+            rating_active = vals['rating_active']
+            task_types = self.env['project.task.type'].search([('rating_active', '=', True)])
+            if (not task_types and rating_active) or (task_types and task_types <= self and not rating_active):
+                mt_project_task_rating = self.env.ref('project.mt_project_task_rating')
+                mt_project_task_rating.hidden = not rating_active
+                mt_project_task_rating.default = rating_active
+                self.env.ref('project.mt_task_rating').hidden = not rating_active
+                self.env.ref('project.rating_project_request_email_template').active = rating_active
         return super().write(vals)
 
     def copy_data(self, default=None):
@@ -164,15 +208,6 @@ class ProjectTaskType(models.Model):
             }
         return res
 
-    @api.depends('project_ids', 'project_ids.rating_active')
-    def _compute_disabled_rating_warning(self):
-        for stage in self:
-            disabled_projects = stage.project_ids.filtered(lambda p: not p.rating_active)
-            if disabled_projects:
-                stage.disabled_rating_warning = '\n'.join('- %s' % p.name for p in disabled_projects)
-            else:
-                stage.disabled_rating_warning = False
-
     @api.depends('project_ids')
     def _compute_user_id(self):
         """ Fields project_ids and user_id cannot be set together for a stage. It can happen that
@@ -185,3 +220,20 @@ class ProjectTaskType(models.Model):
     def _check_personal_stage_not_linked_to_projects(self):
         if any(stage.user_id and stage.project_ids for stage in self):
             raise UserError(_('A personal stage cannot be linked to a project because it is only visible to its corresponding user.'))
+
+    # ---------------------------------------------------
+    # Rating business
+    # ---------------------------------------------------
+
+    # This method should be called once a day by the scheduler
+    @api.model
+    def _send_rating_all(self):
+        stages = self.search([
+            ('rating_active', '=', True),
+            ('rating_status', '=', 'periodic'),
+            ('rating_request_deadline', '<=', fields.Datetime.now())
+        ])
+        for stage in stages:
+            stage.project_ids.task_ids._send_task_rating_mail()
+            stage._compute_rating_request_deadline()
+            self.env.cr.commit()

--- a/addons/project/models/res_config_settings.py
+++ b/addons/project/models/res_config_settings.py
@@ -1,62 +1,18 @@
 # -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
-from odoo import api, fields, models
+from odoo import fields, models
 
 
 class ResConfigSettings(models.TransientModel):
     _inherit = 'res.config.settings'
 
     module_hr_timesheet = fields.Boolean(string="Task Logs")
-    group_project_rating = fields.Boolean("Customer Ratings", implied_group='project.group_project_rating')
     group_project_stages = fields.Boolean("Project Stages", implied_group="project.group_project_stages")
-    group_project_recurring_tasks = fields.Boolean("Recurring Tasks", implied_group="project.group_project_recurring_tasks", group='base.group_portal,base.group_user')
-    group_project_task_dependencies = fields.Boolean("Task Dependencies", implied_group="project.group_project_task_dependencies")
-    group_project_milestone = fields.Boolean('Milestones', implied_group='project.group_project_milestone', group='base.group_portal,base.group_user')
-
-    @api.model
-    def _get_basic_project_domain(self):
-        return []
 
     def set_values(self):
-        # Ensure that settings on existing projects match the above fields
-        projects = self.env["project.project"].search([])
-        basic_projects = projects.filtered_domain(self._get_basic_project_domain())
-
-        features = {
-            # key: (config_flag, is_global), value: project_flag
-            ("group_project_rating", True): "rating_active",
-            ("group_project_task_dependencies", False): "allow_task_dependencies",
-            ("group_project_milestone", False): "allow_milestones",
-        }
-
-        for (config_flag, is_global), project_flag in features.items():
-            config_flag_global = f"project.{config_flag}"
-            config_feature_enabled = self[config_flag]
-            if self.env.user.has_group(config_flag_global) != config_feature_enabled:
-                if config_feature_enabled and not is_global:
-                    basic_projects[project_flag] = config_feature_enabled
-                else:
-                    projects[project_flag] = config_feature_enabled
-
-        task_waiting_subtype_id = self.env.ref('project.mt_task_waiting')
-        project_task_waiting_subtype_id = self.env.ref('project.mt_project_task_waiting')
-        if task_waiting_subtype_id.hidden != (not self['group_project_task_dependencies']):
-            task_waiting_subtype_id.hidden = not self['group_project_task_dependencies']
-            project_task_waiting_subtype_id.hidden = not self['group_project_task_dependencies']
         # Hide Project Stage Changed mail subtype according to the settings
         project_stage_change_mail_type = self.env.ref('project.mt_project_stage_change')
         if project_stage_change_mail_type.hidden == self['group_project_stages']:
             project_stage_change_mail_type.hidden = not self['group_project_stages']
-        # Hide task rating tempalate when customer rating is disbled
-        task_rating_subtype_id = self.env.ref('project.mt_project_task_rating')
-        task_rating_subtype_id.hidden = not self['group_project_rating']
-        self.env.ref('project.mt_task_rating').hidden = not self['group_project_rating']
-        task_rating_subtype_id.default = self['group_project_rating']
-        rating_project_request_email_template = self.env.ref('project.rating_project_request_email_template')
-        if rating_project_request_email_template.active != self['group_project_rating']:
-            rating_project_request_email_template.active = self['group_project_rating']
-        if not self['group_project_recurring_tasks']:
-            self.env['project.task'].sudo().search([('recurring_task', '=', True)]).write({'recurring_task': False})
-
         super().set_values()
